pdns/dnsdistdist/dnsrulactions.cc

```diff
@@ -0,0 +1,87 @@
+#include "dnsrulactions.hh"
+#include <iostream>
+
+using namespace std;
+
+TeeAction::TeeAction(const ComboAddress& ca) : d_remote(ca)
+{
+  d_fd=SSocket(d_remote.sin4.sin_family, SOCK_DGRAM, 0);
+  SConnect(d_fd, d_remote);
+  setNonBlocking(d_fd);
+  d_worker=std::thread(std::bind(&TeeAction::worker, this));  
+}
+
+TeeAction::~TeeAction()
+{
+  d_pleaseQuit=true;
+  close(d_fd);
+  d_worker.join();
+}
+
+DNSAction::Action TeeAction::operator()(DNSQuestion* dq, string* ruleresult) const 
+{
+  if(dq->tcp) 
+    d_tcpdrops++;
+  else {
+    d_queries++;
```

Comment:
Given that this may be called from different threads, perhaps we might want to make `d_tcpdrops` and `d_queries` atomic